### ISSUE TITLE
Training loss

### ISSUE THREAD

**Issue by zewang0729-lang** [phamtrongthang123/drifting_policy#1]
Hello, I would like to ask that although his success rate is increasing, the training loss remains at a very high value during the training process. Is this normal? I would be very grateful if you could reply to me.

<img width="2170" height="726" alt="Image" src="https://github.com/user-attachments/assets/243964e2-9751-4bca-ade6-519e63625869" />

**Comment by phamtrongthang123** [comment]
It is normal to be high, or to be more accurate, constant. This is due to a scaling factor of drifting loss. In the toy Colab example of https://github.com/lambertae/drifting, the author didn't apply the scaling factor, so you see the nice and typical curve. But in practice, the loss is mostly constant due to the scaling factor. Without that scaling factor, it will hurt the performance. 

If you want to apply drifting loss, I would advise you to always have a secondary "loss" where you can see if the error is actually reduced or not. In this case, it is mse_error. 

It doesn't mean that the loss will never reduce though. I don't know when that will happen. For one example, if you use this Gr00t repo (https://github.com/phamtrongthang123/Isaac-GR00T-drifting-loss/) you will see the loss will go down dramatically at the end of training.

**Comment by zewang0729-lang** [comment]
> It is normal to be high, or to be more accurate, constant. This is due to a scaling factor of drifting loss. In the toy Colab example of https://github.com/lambertae/drifting, the author didn't apply the scaling factor, so you see the nice and typical curve. But in practice, the loss is mostly constant due to the scaling factor. Without that scaling factor, it will hurt the performance.
> 
> If you want to apply drifting loss, I would advise you to always have a secondary "loss" where you can see if the error is actually reduced or not. In this case, it is mse_error.
> 
> It doesn't mean that the loss will never reduce though. I don't know when that will happen. For one example, if you use this Gr00t repo (https://github.com/phamtrongthang123/Isaac-GR00T-drifting-loss/) you will see the loss will go down dramatically at the end of training.

Thank you so much for your detailed reply — this has been incredibly helpful to me.
